client/fingerprint_manager.go

```diff
@@ -336,6 +325,18 @@ func (fm *FingerprintManager) fingerprintDriver(name string, f fingerprint.Finge
 	var response cstructs.FingerprintResponse
 
 	fm.nodeLock.Lock()
+
+	// Determine if the driver has been detected before.
+	originalNode, haveDriver := fm.node.Drivers[name]
+	firstDetection := !haveDriver
```

Comment:
Maybe rename to `notPreviouslyDetected` or some variable name that conveys the positive case is a failure to detect the driver.